lib/scholar/decomposition/pca.ex

```diff
@@ -0,0 +1,216 @@
+defmodule Scholar.Decomposition.PCA do
+  @moduledoc """
+  PCA decomposition algorithm.
+  """
+  import Nx.Defn
+
+  @derive {Nx.Container,
+           containers: [
+             :components,
+             :explained_variance,
+             :explained_variance_ratio,
+             :singular_values,
+             :mean,
+             :num_components,
+             :num_features,
+             :num_samples,
+             :decomposer
+           ]}
```

Comment:
If components is not meant to be a tensor, you'll need this:

```suggestion
           keep: [:num_components],
           containers: [
             :components,
             :explained_variance,
             :explained_variance_ratio,
             :singular_values,
             :mean,
             :num_features,
             :num_samples,
             :decomposer
           ]}
```